### PR TITLE
[Feature][REVIEW] Enable DGL cugaph nightly CI 

### PR DESCRIPTION
**Pull request by VibhuJawa**
## Description
This PR closes  https://github.com/dmlc/dgl/issues/4324 .   

Based on discussion on the issue we decided to do `always_pull=True` . 

This uses the docker image present at [link](https://hub.docker.com/r/rapidsai/cugraph_nightly_torch-cuda/tags) which is updated based on the repo: https://github.com/rapidsai/dgl-cugraph-build-environment

Todo:
- [ ] Create a separate Jenkins job for cugraph nightly

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by Rhett-Ying — COMMENTED** [review]
the nightly image will be updated daily? And have you verified cugraph test works on this image?

**Comment by VibhuJawa** [comment]
> the nightly image will be updated daily? And have you verified cugraph test works on this image?

Yea.  The nightly image is updated daily.  This is the [repo](https://github.com/rapidsai/dgl-cugraph-build-environment) that controls it and the image is pushed here twice a [day](https://hub.docker.com/r/rapidsai/cugraph_nightly_torch-cuda/).

**Review by Rhett-Ying — APPROVED** [review]
LGTM

**Comment by VibhuJawa** [comment]
@Rhett-Ying ,  I just want to be sure that before we merge this we dont cause a hard CI dependency on cugraph to always succeed the CI for PRs that dont use this.

**Comment by Rhett-Ying** [comment]
As `Jenkinsfile` is protected and you don't have write permission probably, so the CI triggered by this PR will not apply the change you made in this PR. that's why I hope you could verify it manually.

**Comment by VibhuJawa** [comment]
> As `Jenkinsfile` is protected and you don't have write permission probably, so the CI triggered by this PR will not apply the change you made in this PR. that's why I hope you could verify it manually.

Okay. Let me verify it again to be sure.

**Comment by Rhett-Ying** [comment]
thanks. once verified, pls feel free to merge it.

**Comment by Rhett-Ying** [comment]
@VibhuJawa This PR will break down the CI. I have not found out the root cause. I will fetch your PR and open a new one.

File: Jenkinsfile [hunk 1]
```diff
@@ -296,9 +296,9 @@ pipeline {
               agent {
                 docker {
                   label "linux-cpu-node"
-                  image "nvcr.io/nvidia/pytorch:22.04-py3"
+                  image "rapidsai/cugraph_nightly_torch-cuda:11.5-base-ubuntu18.04-py3.9-pytorch1.11.0-rapids22.10"
                   args "-u root"
-                  alwaysPull false
+                  alwaysPull True
```
Comment (Rhett-Ying) [inline Jenkinsfile:301]: probably should use `true` instead of `True`.

**Pull request merged**